Tasks/Common/webdeployment-common-v2/fileTransformationsUtility.ts

```diff
@@ -54,54 +54,58 @@ export function advancedFileTransformations(isFolderBasedDeployment: boolean, ta
+                var args = ParameterParser.parse(rule);
+                if(Object.keys(args).length < 2 || !args["transform"] || !args["xml"]) {
+                    tl.error(tl.loc("MissingArgumentsforXMLTransformation"));
                 }
-                isTransformationApplied = xdtTransformationUtility.basicXdtTransformation(folderPath, transformConfigs);
-            }
+                else if(Object.keys(args).length > 2) {
+                    isTransformationApplied = xdtTransformationUtility.specialXdtTransformation(folderPath, args["transform"].value, args["xml"].value, args["result"].value) && isTransformationApplied;
+                }
+                else {
+                    isTransformationApplied = xdtTransformationUtility.specialXdtTransformation(folderPath, args["transform"].value, args["xml"].value) && isTransformationApplied;
+                }
+            });
             
             if(isTransformationApplied) {
                 console.log(tl.loc("XDTTransformationsappliedsuccessfully"));
+            }
+            else {
+                tl.error(tl.loc('FailedToApplySpecialTransformation'));
             }            
         }
     }
 
     if(variableSubstitutionFileFormat === "xml") {
+        let isSubstitutionApplied: boolean = true;
         if(targetFiles.length == 0) { 
-            xmlSubstitutionUtility.substituteAppSettingsVariables(folderPath, isFolderBasedDeployment);
+            isSubstitutionApplied = xmlSubstitutionUtility.substituteAppSettingsVariables(folderPath, isFolderBasedDeployment);
         }
         else {            
             targetFiles.forEach(function(fileName) { 
-                xmlSubstitutionUtility.substituteAppSettingsVariables(folderPath, isFolderBasedDeployment, fileName);
+                isSubstitutionApplied = xmlSubstitutionUtility.substituteAppSettingsVariables(folderPath, isFolderBasedDeployment, fileName) || isSubstitutionApplied;
             });
         }
-        console.log(tl.loc('XMLvariablesubstitutionappliedsuccessfully'));
+        if(isSubstitutionApplied) {
+            console.log(tl.loc('XMLvariablesubstitutionappliedsuccessfully')); 
+        } 
+        else { 
+            tl.error(tl.loc('FailedToApplyXMLvariablesubstitution')); 
+        }
     }
 
     if(variableSubstitutionFileFormat === "json") {
         // For Json variable substitution if no target files are specified file files matching **\*.json
         if(!targetFiles || targetFiles.length == 0) {
             targetFiles = ["**/*.json"];
         }
-        jsonSubstitutionUtility.jsonVariableSubstitution(folderPath, targetFiles, true);
-        console.log(tl.loc('JSONvariablesubstitutionappliedsuccessfully'));
+        let isSubstitutionApplied = jsonSubstitutionUtility.jsonVariableSubstitution(folderPath, targetFiles, true);
+        if(isSubstitutionApplied) {
+            console.log(tl.loc('XMLvariablesubstitutionappliedsuccessfully')); 
```

Comment:
Here check is for JSON variable substitution through success and failure messages are for XML variable substituion